services/autoscaling/src/simcore_service_autoscaling/background_task.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+import contextlib
+import datetime
+import logging
+from typing import Awaitable, Callable
+
+from servicelib.logging_utils import log_context
+
+logger = logging.getLogger(__name__)
+
+
+async def _periodic_scheduled_task(
+    task: Callable[..., Awaitable[None]],
+    *,
+    interval: datetime.timedelta,
+    task_name: str,
+    **kwargs,
+):
+    while await asyncio.sleep(interval.total_seconds(), result=True):
+        try:
+            with log_context(logger, logging.DEBUG, msg=f"Run {task_name}"):
```

Comment:
nice reference!
But I do not think that it is useful to log everytime that task runs, since 90% of the time it will do nothing, so I will keep this as DEBUG wher it can be interesting in case of very specific debugging or testing.